### PR TITLE
[Merged by Bors] - feat: Double quotient of an algebra

### PR DESCRIPTION
**Pull request by YaelDillies**
Match https://github.com/leanprover-community/mathlib/pull/18452

[`ring_theory.ideal.quotient_operations`@`d3acee0d776b15ffb8318f327325ff343cc8bdcc`..`b88d81c84530450a8989e918608e5960f015e6c8`](https://leanprover-community.github.io/mathlib-port-status/file/ring_theory/ideal/quotient_operations?range=d3acee0d776b15ffb8318f327325ff343cc8bdcc..b88d81c84530450a8989e918608e5960f015e6c8)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Vierkantor** [comment]
This is one of the few out of sync files I can do something about, so let me try working on this for an hour (and maybe a bit more later).

File: Mathlib/RingTheory/Ideal/QuotientOperations.lean [hunk 1]
```diff
@@ -670,4 +670,194 @@ theorem quotQuotEquivComm_algebraMap (x : R) :
...
+
+@[simp]
+theorem quotQuotEquivComm_comp_quotQuotMkₐ :
+    AlgHom.comp (↑(quotQuotEquivCommₐ R I J)) (quotQuotMkₐ R I J) = quotQuotMkₐ R J I :=
+  AlgHom.ext <| quotQuotEquivComm_quotQuotMk I J
+#align double_quot.quot_quot_equiv_comm_comp_quot_quot_mkₐ DoubleQuot.quotQuotEquivComm_comp_quotQuotMkₐ
+
+variable {I J}
+
+/-- The **third isomorphism theorem** for rings. See `quotQuotEquivQuotSupₐ` for version
```
Comment (wupr) [inline Mathlib/RingTheory/Ideal/QuotientOperations.lean:817]: "Rings" should read "algebras". This is carried over from an upstream error I made in Lean 3 mathlib.

File: Mathlib/RingTheory/Ideal/QuotientOperations.lean [hunk 2]
```diff
@@ -670,4 +670,194 @@ theorem quotQuotEquivComm_algebraMap (x : R) :
...
+
+@[simp]
+theorem quotQuotEquivQuotOfLeₐ_toRingEquiv (h : I ≤ J) :
+    (quotQuotEquivQuotOfLeₐ R h : _ ⧸ J.map (Quotient.mkₐ R I) ≃+* _) = quotQuotEquivQuotOfLe h :=
+  rfl
+#align double_quot.quot_quot_equiv_quot_of_leₐ_to_ring_equiv DoubleQuot.quotQuotEquivQuotOfLeₐ_toRingEquiv
+
+@[simp]
+-- Porting note: had to add an extra coercion arrow on the right hand side.
+theorem coe_quotQuotEquivQuotOfLeₐ (h : I ≤ J) :
```
Comment (ChrisHughes24) [inline Mathlib/RingTheory/Ideal/QuotientOperations.lean:831]:
```suggestion
theorem coe_quotQuotEquivQuotOfLEₐ (h : I ≤ J) :
```
Here and everywhere else

**Comment by ChrisHughes24** [comment]
bors merge

**Pull request closed without merging**